emulate/emulate_queue.py

```diff
@@ -163,21 +166,60 @@ def retrieve_next_animation() -> Code:
 		return code_from_priority_queue
 	return retrieve_random_animation()
 
+
 def reset_dmx_values():
 	number_of_values = 5 * 28 * 3  # 5 rows, 28 columns, 3 color values
 	serialized = ",".join("0" * number_of_values)
 	redis_db.set('DMXvalues', serialized)
 
 
+def start_udp_server():
+	UDPServerSocket = socket.socket(socket.AF_INET, type=socket.SOCK_DGRAM)
+	UDPServerSocket.setblocking(0)
+	UDPServerSocket.bind(("0.0.0.0", int(os.environ.get('UDP_SERVER_PORT', 20001))))
+	while Config.objects.first().udp_receive_run:
+		try:
+			bytesAddressPair = UDPServerSocket.recvfrom(2048)
+
+			message = bytesAddressPair[0]
+			message = message.decode('utf-8')
+			message = to_json(message)
+
+			if message and str(message.get("key")) == str(Config.objects.first().udp_key):
+				serialized = ",".join([str(x) for x in list(message["stage"])])
+				redis_db.set('DMXvalues', serialized)
+		except Exception as e:
```

Comment:
Lepiej gdzieś zlogować ten błąd.

Reply:
to generalnie cały czas będzie loggować błędy w momencie gdy nie ma pakietu

 `bytesAddressPair = UDPServerSocket.recvfrom(2048)`

inaczej cały kod zawiesiłby się na tej linii czekając na pakiet